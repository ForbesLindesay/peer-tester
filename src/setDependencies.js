function setDependencies(pkg, dependencies) {
  Object.keys(dependencies).forEach(name => {
    if (pkg.dependencies && (name in pkg.dependencies)) {
      throw new Error(
        name + ' is in both peerDependencies and dependencies. ' +
        'Peer dependencies should not be listed as dependencies.'
      );
    }
  });
  return {
    ...pkg,
    peerDependencies: dependencies,
    devDependencies: {
      ...(pkg.devDependencies || {}),
      ...dependencies,
    },
  };
}

export default setDependencies;
