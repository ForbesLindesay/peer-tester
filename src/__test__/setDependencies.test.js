import setDependencies from '../setDependencies';

test('setDependencies({"peerDependencies": {"react": "^0.14.0"}, "devDependencies": {"react": "^0.14.0"}}, {"react": "0.14.0"})', async () => {
  const dependecies = setDependencies(
    {"peerDependencies": {"react": "^0.14.0"}, "devDependencies": {"react": "^0.14.0"}},
    {"react": "0.14.0"},
  );
  expect(dependecies).toMatchSnapshot();
});
test('setDependencies({"dependencies": {"react": "^0.14.0"}}, {"react": "0.14.0"})', async () => {
  expect(() => setDependencies(
    {"dependencies": {"react": "^0.14.0"}},
    {"react": "0.14.0"},
  )).toThrow();
});
