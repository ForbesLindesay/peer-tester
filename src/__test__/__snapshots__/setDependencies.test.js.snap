// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setDependencies({"peerDependencies": {"react": "^0.14.0"}, "devDependencies": {"react": "^0.14.0"}}, {"react": "0.14.0"}) 1`] = `
Object {
  "devDependencies": Object {
    "react": "0.14.0",
  },
  "peerDependencies": Object {
    "react": "0.14.0",
  },
}
`;
